File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int[] count = new int[2];
        
        for (int i : students) count[i]++;
        
        int rem = sandwiches.length;
        
        for (int i : sandwiches){
            if (count[i] == 0) break;
            if (rem-- == 0) break;
            count[i]--;
        }
        return rem;
    }
}

File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        for i in range(len(sandwiches)):
            if students == []:
                return 0
            
            step = 0
            while sandwiches[i] != students[0] and step < len(students):
                l = students.pop(0)
                students.append(l)
                step += 1
                
            if students[0] == sandwiches[i]:
                students.pop(0)
            else:
                return len(students)
        return len(students)